policies/booking_policy.rb

```diff
@@ -0,0 +1,19 @@
+class BookingPolicy
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+
+  def initialize(params)
+    @user_email   = params[:user_email]
+    @first_class  = params[:first_class] || 0
+    @second_class = params[:second_class] || 0
+  end
+
+  def params_valid?
+    (@first_class.to_i > 0 || @second_class.to_i > 0) && valid_email?
```

Comment:
Do not use instance variables, use `attr_reader` & make it private for encapsulation 
For example
```ruby
class Example
  def initialize(params)
    @a = params[:a]
  end
  
  private
  
  attr_reader :a
end
```